include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -96,26 +96,24 @@ namespace alpaka
             template<typename TDim, typename TIdx>
             ALPAKA_FN_HOST auto checkVecOnly3Dim(Vec<TDim, TIdx> const& vec) -> void
             {
-                for(auto i = std::min(static_cast<typename TDim::value_type>(3), TDim::value); i < TDim::value; ++i)
-                {
-                    if(vec[TDim::value - 1u - i] != 1)
-                    {
-                        throw std::runtime_error("The CUDA/HIP accelerator supports a maximum of 3 dimensions. All "
-                                                 "work division extents of the dimensions higher 3 have to be 1!");
-                    }
-                }
+                if constexpr(TDim::value > 0)
+                    for(auto i = std::min(typename TDim::value_type{3}, TDim::value); i < TDim::value; ++i)
+                        if(vec[TDim::value - 1u - i] != 1)
+                            throw std::runtime_error(
+                                "The CUDA/HIP accelerator supports a maximum of 3 dimensions. All "
+                                "work division extents of the dimensions higher 3 have to be 1!");
```

Comment:
Please add a few `{}` here.

Reply:
> parentheses should be omitted if the body is a single statement:

Parentheses should be omitted if the body is a *one liner*, not a single statement. I don't think the construction above counts as one line...

Reply:
I find that fragile. That would mean if I would refactor a variable name and clang-format then decides to break, that I would need to add braces as well.